app/V1Module/presenters/ReferenceExerciseSolutionsPresenter.php

```diff
@@ -103,7 +103,14 @@ public function actionExercise(string $exerciseId) {
       throw new ForbiddenRequestException("You cannot access this exercise solutions");
     }
 
-    $this->sendSuccessResponse($exercise->getReferenceSolutions()->getValues());
+    $refSolutions = array_map(function($rs) use($exercise) {
+      $json = $rs->jsonSerialize();
+      $json['canDelete'] = $this->exerciseAcl->canDeleteReferenceSolution($exercise, $rs);
```

Comment:
I seem to have missed this yesterday. Does the `canDeleteReferenceSolution` method really need the exercise parameter? The exercise could be acquired with `$rs->getExercise()` in permission policy methods...

Reply:
Creating another endpoint, which would also be a temporary solution, does not make any sense. Especially, if it does not bring anything good (only a philosophical improvement on which we do not have consensus), means a lot of work (which will be dismissed later), and will increase the amount of requests (one per each button) not mentioning worsening UX. I think that the right course of action is either accept this solution as temporary, or go for the right solution (but it is needed really soon).

Reply:
Ugh, wrong thread. Yes, the `canDeleteReferenceSolution` can be simplified. Will do.